src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -43,8 +47,8 @@ const useDiscovery = () => {
     return newTopicLists[indexTopic];
   };
 
-  const mapUser = (newUser) => {
-    return discovery.initialUsers.map((user) => {
+  const mapUserWith = (listReference, newUser) => {
+    return listReference.map((user) => {
       if (user.user) {
         if (user.user.user_id === newUser.user.user_id) return newUser;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function name `mapUserWith` is more descriptive than the previous `mapUser`, as it now clearly indicates that it maps a user with a reference list. This improves readability and understanding of the code.



<!-- This is an auto-generated comment by OSS CodeRabbit -->